dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/trackedentity/TrackedEntityAttributeStoreIntegrationTest.java

```diff
@@ -0,0 +1,212 @@
+        trackedEntityTypeService.updateTrackedEntityType( trackedEntityTypeB );
+
+        programB = createProgram( 'B' );
+        programService.addProgram( programB );
+
+        List<ProgramTrackedEntityAttribute> pteaList = IntStream.range( A, T )
+            .mapToObj( i -> Character.toString( (char) i ) ).map( s -> new ProgramTrackedEntityAttribute( programB,
+                attributeService.getTrackedEntityAttributeByName( "Attribute" + s ) ) )
+            .collect( Collectors.toList() );
+
+        // Setting searchable to true for 5 program tracked entity attributes
+        ProgramTrackedEntityAttribute ptea = pteaList.get( 0 );
+        ptea.setSearchable( true );
+        ptea = pteaList.get( 4 );
+        ptea.setSearchable( true );
+        ptea = pteaList.get( 9 );
+        ptea.setSearchable( true );
+        ptea = pteaList.get( 13 );
+        ptea.setSearchable( true );
+        ptea = pteaList.get( 18 );
+        ptea.setSearchable( true );
+
+        programB.getProgramAttributes().addAll( pteaList );
+        programService.updateProgram( programB );
+
+    }
+
+    @Test
+    public void testGetAllIndexableAttributes()
+    {
+        long idA = attributeService.addTrackedEntityAttribute( attributeW );
+        long idB = attributeService.addTrackedEntityAttribute( attributeY );
+
+        Set<TrackedEntityAttribute> indexableAttributes = attributeService
+            .getAllTrigramIndexableTrackedEntityAttributes();
+
+        assertNotNull( indexableAttributes );
+        assertEquals( indexableAttributes.size(), 9 );
+        assertTrue( indexableAttributes.contains( attributeW ) );
+        assertTrue( indexableAttributes.contains( attributeY ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'A' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'E' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'J' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'N' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'O' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'S' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'T' ) ) );
+    }
+
+    @Test
+    public void testCreateTrigramIndex()
+    {
+        long idA = attributeService.addTrackedEntityAttribute( attributeW );
+        attributeService.createTrigramIndex( attributeW );
+    }
+
+    @Test
+    public void testRunAnalyze()
+    {
+        attributeService.runAnalyze();
```

Comment:
same here and below. not sure if there is something we can assert

Reply:
asserted by rerunning analyze directly with jdbcTemplate. doesnt make much sense tbh. 🤔  Or I should just drop the assertion part (which Sonar complains about then). Otherwise I should change the void return type to boolean and assert that it returns true. What do you think?

Reply:
@teleivo  what do you think of the above comment?